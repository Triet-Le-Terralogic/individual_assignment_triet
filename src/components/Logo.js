import React from "react";
import logo from "../assets/img/brand_logo.svg";

export default function Logo() {
  return (
    <div>
      <img src={logo} alt="Brand logo" />
    </div>
  );
}
