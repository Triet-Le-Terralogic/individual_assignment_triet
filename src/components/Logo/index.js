import React from "react";
import logo from "../../assets/img/brand_logo.svg";

export default function Logo() {
	return (
		<div className="Logo text-center text-lg-left">
			<img src={logo} alt="Brand logo" />
		</div>
	);
}
