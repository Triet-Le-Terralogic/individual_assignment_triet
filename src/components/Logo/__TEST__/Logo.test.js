import React from "react";
import Logo from "../index";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const wrapper = shallow(<Logo />);
let container, containerProps;

describe("<Logo/>", () => {
  beforeEach(() => {
    container = wrapper.find("div");
    containerProps = container.props();
  });

  it("should render without crash", () => {
    shallow(<Logo />);
  });

  it("should render correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render a div with class: Logo", () => {
    const container = wrapper.find("div");
    expect(container).toHaveLength(1);
    expect(containerProps.className).toContain("Logo");
  });

  it("should render a img with default imported src", () => {
    const image = container.find("img");
    expect(image).toHaveLength(1);
    expect(image.props().src).toEqual("brand_logo.svg");
  });
});
