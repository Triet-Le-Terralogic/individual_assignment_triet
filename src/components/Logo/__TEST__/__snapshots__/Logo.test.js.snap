// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo/> should render correctly 1`] = `
<div
  className="Logo text-center text-lg-left"
>
  <img
    alt="Brand logo"
    src="brand_logo.svg"
  />
</div>
`;
